flamedisx/likelihood.py

```diff
@@ -149,21 +155,54 @@ def set_data(self,
         update specific datasets.
         """
         if isinstance(data, pd.DataFrame):
-            # Only one dataset
             assert len(self.dsetnames) == 1, \
-                "You passed a DataFrame but there are multiple datasets"
+                "You passed one DataFrame but there are multiple datasets"
             data = {DEFAULT_DSETNAME: data}
 
+        if any([d is None for d in data.values()]):
+            # At least one dataset is None => reset everything!
+            # TODO: should this give a warning?
```

Comment:
Yes, also what does reset mean here. If all datasets are set to None we can only use the Source for simulation but a user doesn't first need to reset the Source to do that, so why support it. I guess it's less prone to subtle bugs if all data is set to None instead of one dataset, so without adding a warning this makes sense.